### PR TITLE
[Mobile app] Code cleanup

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Removes invertase due to iOS build failure.
- [x] Upgrades pod version.
- [x] Replaces relative imports with absolute imports.
- [x] Updates Google maps version for iOS.
- [x] Configures recommended code metrics

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [ticket_id]()

#### Screenshots (optional)

### CODE REVIEW

File: mobile/analysis_options.yaml [hunk 1]
```diff
@@ -1,179 +1,114 @@
...
-    - test/**
+    maintainability-index: 50
+    technical-debt:
+      threshold: 1
+      todo-cost: 161
+      ignore-cost: 320
+      ignore-for-file-cost: 396
+      as-dynamic-cost: 322
+      deprecated-annotations-cost: 37
+      file-nullsafety-migration-cost: 41
```
Comment (NoahNsimbe) [inline mobile/analysis_options.yaml:43]: Added more code metrics

File: mobile/ios/Podfile [hunk 2]
```diff
@@ -34,7 +34,7 @@ target 'Runner' do
   use_modular_headers!
 
   flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
-  pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '9.5.0'
+#   pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '10.0.0'
```
Comment (NoahNsimbe) [inline mobile/ios/Podfile:37]: Disabling invertase due to build failure on iOS

File: mobile/lib/models/enum_constants.dart [hunk 3]
```diff
@@ -300,56 +302,46 @@ enum Frequency {
...
+                  color: charts.ColorUtil.fromDartColor(CustomColors.greyColor),
                 ),
               ),
-            ),
-          );
-        }
-
-        return dailyTicks;
+            )
+            .toList();
```
Comment (NoahNsimbe) [inline mobile/lib/models/enum_constants.dart:315]: replacing for loop with a map function

File: mobile/lib/models/enum_constants.dart [hunk 4]
```diff
@@ -425,15 +417,15 @@ enum Pollutant {
...
           return AirQuality.ufsgs;
-        } else if (value.isWithin(251.00, 350.99)) {
+        } else if (value.isWithin(251, 350.99)) {
           return AirQuality.unhealthy;
-        } else if (value.isWithin(351.00, 430.99)) {
+        } else if (value.isWithin(351, 430.99)) {
           return AirQuality.veryUnhealthy;
-        } else if (value >= 431.00) {
+        } else if (value >= 431) {
           return AirQuality.hazardous;
```
Comment (NoahNsimbe) [inline mobile/lib/models/enum_constants.dart:429]: double literal formatting.

File: mobile/lib/screens/auth/phone_auth_widget.dart [hunk 5]
```diff
@@ -2,21 +2,15 @@ import 'dart:async';
...
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 
-import '../../services/firebase_service.dart';
-import '../../themes/app_theme.dart';
-import '../../themes/colors.dart';
-import '../../utils/network.dart';
-import '../../widgets/custom_shimmer.dart';
-import '../../widgets/custom_widgets.dart';
 import '../on_boarding/profile_setup_screen.dart';
```
Comment (NoahNsimbe) [inline mobile/lib/screens/auth/phone_auth_widget.dart:14]: reducing import statements

File: mobile/lib/screens/on_boarding/splash_screen.dart [hunk 6]
```diff
@@ -158,17 +151,28 @@ class SplashScreenState extends State<SplashScreen> {
...
+              semanticsLabel: 'Share',
+              // height: 118,
+              // width: 81,
+            ),
+          ],
+        ),
+      ),
     );
   }
 }
```
Comment (NoahNsimbe) [inline mobile/lib/screens/on_boarding/splash_screen.dart:178]: Substituting widget functions with stateless widgets

File: mobile/lib/services/firebase_service.dart [hunk 7]
```diff
@@ -608,27 +606,27 @@ class CustomAuth {
...
-        await showSnackBar(
+        showSnackBar(
           context,
           'Account has been disabled. PLease contact support',
         );
       } else {
-        await showSnackBar(
+        showSnackBar(
           context,
           Config.appErrorMessage,
```
Comment (NoahNsimbe) [inline mobile/lib/services/firebase_service.dart:631]: removing unnecessary await

File: mobile/lib/models/enum_constants.dart (hunk 3 above)
Comment (Baalmart) [reply mobile/lib/models/enum_constants.dart:315]: looks neat, thanks

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe , the PR review comments were exciting. Cant wait to learn more as I onboard to the mobile codebase!

Happy refactoring! :)

**Pull request merged**